frontend/src/components/cars-categories/cars-categories.tsx

```diff
@@ -13,7 +19,30 @@ import 'swiper/css/mousewheel';
 import styles from './styles.module.scss';
 
 const CarsCategories: FC = () => {
-  const [brands] = useState<Brand[]>(mockBrands);
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const { brandDetails } = useAppSelector((state) => state.carFilter);
+  const { id: detailId } = brandDetails[0];
+  const { data: brands } = useGetBrandsQuery();
+  const [search, filteredCars] = useLazyGetFilteredCarsQuery();
+
+  useEffect(() => {
+    if (filteredCars.data) {
+      dispatch(setCars(filteredCars.data));
+    }
+  }, [filteredCars]);
+
+  const doSearch = async (queryParams: string[][]): Promise<void> => {
+    dispatch(
+      setBrandDetailsValue({
+        id: detailId,
+        brandId: queryParams[0][1] || '',
+        modelIds: [],
+      }),
+    );
+    await search(queryParams);
+    navigate(API.SEARCH);
+  };
```

Comment:
That's wrong. **The search page should be responsible for searching cars**. The categories slider should just provide search params for it